pkg/api/validation/events.go

```diff
@@ -25,13 +29,35 @@ import (
 // ValidateEvent makes sure that the event makes sense.
 func ValidateEvent(event *api.Event) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
-	// TODO: There is no namespace required for node.
-	if event.InvolvedObject.Kind != "Node" &&
-		event.Namespace != event.InvolvedObject.Namespace {
-		allErrs = append(allErrs, errs.NewFieldInvalid("involvedObject.namespace", event.InvolvedObject.Namespace, "namespace does not match involvedObject"))
+
+	nonNamespacedKindFlag, err := isNonNamespacedKind(event.InvolvedObject.Kind, event.InvolvedObject.APIVersion)
+	if err != nil {
+		allErrs = append(allErrs, errs.NewFieldInvalid("involvedObject.Kind", event.InvolvedObject.Kind, fmt.Sprintf("couldn't check whether namespace is allowed: %s", err)))
```

Comment:
Hm. What about events about objects from 3rd parties? @smarterclayton does OpenShift make k8s events about its own objects?

(@feihujiang I'm sorry I keep making this so complicated!)